src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,84 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = clazz.getDeclaredFields();
+        if (declaredFields.length == 0
+                || Arrays.stream(declaredFields)
+                .filter(f -> f.getType().isPrimitive())
+                .count() == declaredFields.length) {
+            return createNewInstance(clazz);
+        }
```

Comment:
why do you need this?

Reply:
It didn't work without it. Objects were null if they hadn't any fields or came without any fields if their fields were primitive. I spent a lot of time to find why and this helped.